packages/ckb/src/utils/ckb-tx.ts

```diff
@@ -234,3 +245,58 @@ export const checkCkbTxInputsCapacitySufficient = async (
     .reduce((prev, current) => prev + current, BigInt(0));
   return sumInputsCapacity > sumOutputsCapacity;
 };
+
+export function signCkbTransaction(
+  key: string,
+  ckbTx: CKBComponents.RawTransactionToSign,
+  inputCells?: { outPoint: CKBComponents.OutPoint; lock: CKBComponents.Script }[],
+  skipMissingKeys?: boolean,
+): CKBComponents.RawTransaction;
+
+export function signCkbTransaction(
+  key: Map<string, string>,
+  ckbTx: CKBComponents.RawTransactionToSign,
+  inputCells: { outPoint: CKBComponents.OutPoint; lock: CKBComponents.Script }[],
+  skipMissingKeys?: boolean,
+): CKBComponents.RawTransaction;
+
+export function signCkbTransaction(
+  key: string | Map<string, string>,
+  ckbTx: CKBComponents.RawTransactionToSign,
+  inputCells: { outPoint: CKBComponents.OutPoint; lock: CKBComponents.Script }[] = [],
+  skipMissingKeys = false,
+) {
+  if (key instanceof Map && inputCells.length === 0) {
+    throw new Error('inputCells must not be empty when using Map of keys');
+  }
+
+  const transactionHash = rawTransactionToHash(ckbTx);
+  const signedWitnesses = signWitnesses(key)({
+    transactionHash,
+    witnesses: ckbTx.witnesses,
+    inputCells,
+    skipMissingKeys,
+  });
+
+  return {
+    ...ckbTx,
+    witnesses: signedWitnesses.map((witness) =>
+      typeof witness !== 'string' ? serializeWitnessArgs(witness) : witness,
+    ),
+  };
+}
+
+export const addressToScriptHash = (address: string) => {
+  const script = addressToScript(address);
+  return scriptToHash(script);
+};
+
+// Normalizes a CKB transaction to be signed by ensuring all witnesses are in serialized string format.
+export const normalizeCkbTxToSign = (ckbTx: CKBComponents.RawTransactionToSign): CKBComponents.RawTransactionToSign => {
```

Comment:
Is there any function calling function `normalizeCkbTxToSign`?

Reply:
Not in the current codebase. This function was implemented to ensure that `RawTransactionToSign.witness` is consistently represented as a string, simplifying deserialization in other strongly typed languages.

Reply:
Sorry, I don't understand. 

Why do we keep this function if no other functions call the `normalizeCkbTxToSign`? And the `signCkbTransaction` already has the same code.

If you want a reference for other programming languages ​​on how to do it, adding more comments in the `signCkbTransaction` function can work too.

Reply:
Agreed! Adding comments in `signCkbTransaction` would be a better approach.